src/org/labkey/test/tests/DataClassTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2011-2019 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.labkey.test.tests;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.categories.DailyC;
+import org.labkey.test.components.domain.DomainFormPanel;
+import org.labkey.test.pages.experiment.CreateDataClassPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.PortalHelper;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.labkey.test.util.DataRegionTable.DataRegion;
+
+@Category({DailyC.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 5)
+public class DataClassTest extends BaseWebDriverTest
```

Comment:
@labkey-tchad @labkey-danield FYI, I added an initial DataClassTest here with some test cases to error validation checks on the new data class designer UI. There is a TODO in here and story added to the backlog for adding more UI cases for this designer.